tests/components/primitive-components/source-group-id-in-source-component.test.tsx

```diff
@@ -0,0 +1,243 @@
+import { expect, test } from "bun:test"
+import { getTestFixture } from "tests/fixtures/get-test-fixture"
+
+test("group id present in pcb_component, schematic_component and source_component", async () => {
+  const { circuit } = getTestFixture()
+
+  circuit.add(
+    <board>
+      <group name="G1">
+        <resistor name="R1" footprint={"0402"} resistance={100} />
+        <resistor name="R2" footprint={"0402"} resistance={100} />
+      </group>
+      <capacitor name="C1" footprint={"0402"} capacitance={100} />
+    </board>,
+  )
+
+  circuit.renderUntilSettled()
+
+  const schComponents = circuit.db.schematic_component.list()
+  expect(schComponents).toMatchInlineSnapshot(`
```

Comment:
this snapshot is going to change too frequently, the main point is to test the schematic groups. We can reduce the test to just test for schematic group snapshots

Reply:
you can also test that the schematic components have a schematic group id